shims/common/src/main/scala/io/glutenproject/GlutenConfig.scala

```diff
@@ -1324,6 +1326,13 @@ object GlutenConfig {
       .booleanConf
       .createWithDefault(true)
 
+  val ENABLE_COLUMNAR_PROJECT_COLLAPSE =
+    buildConf("spark.gluten.sql.columnar.project.collapse")
+      .internal()
+      .doc("Combines two columnar project operators into one and perform alias substitution")
+      .booleanConf
+      .createWithDefault(false)
```

Comment:
We should enable it by default and pass CI to see if there are some bad case.

Reply:
Originally it was enabled by default, but the unit test for ck failed because some tests in ck directly used the index to retrieve the plan. After my optimization, the number of plans decreased, causing an IndexOutOfBoundsException in the unit test for ck. If it's just for validation, it can be enabled by default and then disabled later.
```
00:39:58  16:39:58.131 WARN io.glutenproject.metrics.MetricsUtil: Updating native metrics failed due to the wrong size of metrics data: 7
00:39:58  - Check TPCH Q2 metrics updater *** FAILED ***
00:39:58    java.lang.IndexOutOfBoundsException: 10
00:39:58    at scala.collection.mutable.ResizableArray.apply(ResizableArray.scala:46)
00:39:58    at scala.collection.mutable.ResizableArray.apply$(ResizableArray.scala:45)
00:39:58    at scala.collection.mutable.ArrayBuffer.apply(ArrayBuffer.scala:49)
00:39:58    at io.glutenproject.execution.metrics.GlutenClickHouseTPCHMetricsSuite.$anonfun$new$9(GlutenClickHouseTPCHMetricsSuite.scala:210)
00:39:58    at io.glutenproject.execution.metrics.GlutenClickHouseMetricsUTUtils$.executeMetricsUpdater(GlutenClickHouseMetricsUTUtils.scala:117)
00:39:58    at io.glutenproject.execution.metrics.GlutenClickHouseTPCHMetricsSuite.$anonfun$new$7(GlutenClickHouseTPCHMetricsSuite.scala:205)
00:39:58    at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
00:39:58    at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
00:39:58    at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
00:39:58    at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
```

Reply:
We can just fix it, e.g., change 10 to 9 if we have collapsed two projects into one. I think the code change should be test only.